### ISSUE TITLE
Add a Code of Conduct

### ISSUE THREAD

**Issue by jayaddison** [hhursev/recipe-scrapers#1341]
I've procrastinated about opening this feature request, because generally the community of `recipe-scrapers` has been well-behaved and non-disruptive, and also I don't relish the additional procedural and emotional burden of dealing with any code-of-conduct violation reports and/or occurrences -- nor do I have much experience doing so.  Despite those reservations, I think it's better to have a Code of Conduct in place; if we fail to live up to it, then that's a more transparent set of circumstances than not having one in the first place (I don't think there's much excuse nowadays).

**Comment by jayaddison** [comment]
I'm starting to do some research into the available options here.

Some material I've begun reading includes:

  * [Initial guidance from GitHub on how to add a code-of-conduct](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project)
  * The [`opensource.guide` Code of Conduct documentation](https://opensource.guide/code-of-conduct/)
    * This quote from the (sadly now defunct) Ada Initiative is important: _"A code of conduct that isn’t (or can’t be) enforced is worse than no code of conduct at all: it sends the message that the values in the code of conduct aren’t actually important or respected in your community."_
    * Currently I don't have -- nor necessarily want -- all the privileges required to enforce code-of-conduct related to issues and pull requests in this repository.  We would need at least one person, preferably multiple (maybe including myself, maybe not) with those permissions if we're going to effectively enforce any policy.
    * This documentation also helps to explain that a scope for the code-of-conduct is important.  `recipe-scrapers` doesn't have (nor is likely to have) in-person events, so I think we can constrain this to GitHub issues and pull requests (and perhaps in future, GitHub discussions also).
  * The [Stumptown Syndicate's incident response resources](https://github.com/stumpsyn/incident-response/blob/master/resources/resources.md).
    * Some of the resources here are intended for in-person events, but much of the guidance is somewhat-universal as it relates to dealing with interpersonal conflicts within communities.

The [Contributor Covenant](https://contributor-covenant.org/) homepage appears unavailable at the moment, but I have read it before and will attempt to read it again to refresh my understanding of it.

:warning: Although not directly policy-related: I have myself verbally assaulted a person (fairly heatedly, certainly not excusable as jest or similar) about a decade ago, and since then I think I've arguably continued to be abusive to that person -- often in what I've rationalized as attempts to apologize or regain what felt like an incredibly important relationship to me -- but that in hindsight may indicate some kind of behavioural problem I can enter into under certain circumstances that I'm still figuring out.

I'm trying to deal with that, and some of the reading I've done for this issue is helpful to me (for example, [this `geekfeminism` article section regarding apologies](https://geekfeminism.fandom.com/wiki/Conference_anti-harassment/Responding_to_reports)) -- but I think it still affects my own sense of credibility as a code-of-conduct policy enforcer.  I would want to do it to make a community better, and would try my best, but I might struggle with some aspects of it and/or become distracted and biased by my own problems.

(sometimes I see people attempt to explain-away or deflect behaviour like that as one-off occurrences by a person; in my case I think I also had some problematic behaviour with someone -- although not to such a serious extent -- a couple of years ago, in relation to a desire for a professional relationship.  So, I wouldn't want to initially accept supporting arguments, despite being well-meaning, that claim "I'm not normally like that" - it might be true, but it doesn't address the problem)

So in summary I think the status is:

  - [x] Determine a scope for a code-of-conduct (answer: GitHub issues, pull requests and discussions)
  - [ ] Determine the best-fit code-of-conduct for this project; deriving or adapting an existing template if necessary.
    * Note: there is a useful [comparison of existing code-of-conducts] at https://geekfeminism.fandom.com/wiki/Code_of_conduct_evaluations
  - [ ] Figure out who would be willing and available to enforce a policy (I am, although with some of the personal concerns / disclaimers noted here).
  - [ ] Document and begin applying the policy (hopefully any reports/actions should be rare; even so, the policy and processes for it need to be in place first).

**Comment by jayaddison** [comment]
> The [Contributor Covenant](https://contributor-covenant.org/) homepage appears unavailable at the moment, but I have read it before and will attempt to read it again to refresh my understanding of it.

Ah, that hyperlink is incorrect; adding the `www.` subdomain prefix opens the Contributor Covenant as expected: https://www.contributor-covenant.org/

**Comment by jayaddison** [comment]
> * Currently I don't have -- nor necessarily want -- all the privileges required to enforce code-of-conduct related to issues and pull requests in this repository.  We would need at least one person, preferably multiple (maybe including myself, maybe not) with those permissions if we're going to effectively enforce any policy.

Although I can currently lock discussion threads and could issue corrections and warnings in relation to policy violations, I think that a `moderator` role would be the most appropriate way to provide permissions for temporary and permanent bans, as required by some of the more popular codes of conduct.  However, to my knowledge, moderator permissions are currently available in GitHub only on organization accounts.  I'm not suggesting any changes or actions there, just noting it to help understand the available options.

**Comment by jayaddison** [comment]
> * [ ]  Figure out who would be willing and available to enforce a policy (I am, although with some of the personal concerns / disclaimers noted here).

I'd like to pause for another month or so before confirming whether I'm genuinely on-board for signing up to this, if/when a policy is put in place.

**Comment by jayaddison** [comment]
Although a Code of Conduct may make sense for this project at some point, this thread contains some personal discussion that make it a non-ideal place to host effective discussion for a policy.

So: I will close this thread; if someone is interested in requesting and/or presenting a Code of Conduct in future, then I'd recommend opening a fresh issue to focus the conversation on the requirements and suggestions for how to implement it.
